test/decompress.js

```diff
@@ -0,0 +1,141 @@
+'use strict'
+
+const { test } = require('node:test')
+const assert = require('node:assert')
+const zlib = require('node:zlib')
+const { decompress } = require('../lib/core/util')
+
+const url = 'http://localhost/'
+
+test('ignores responses with null body', () => {
+  const body = decompress(new Request(url), new Response(null))
+  assert.equal(body, null)
+})
+
+test('ignores responses without the "Content-Encoding" header', async () => {
+  const body = decompress(new Request(url), new Response('hello world'))
+  assert.equal(await new Response(body).text(), 'hello world')
+})
+
+test('ignores responses with empty "Content-Encoding" header', async () => {
+  const body = decompress(
+    new Request(url),
+    new Response('hello world', {
+      headers: { 'content-encoding': '' }
+    })
+  )
+  assert.equal(await new Response(body).text(), 'hello world')
+})
+
+test('ignores redirect responses', async () => {
+  const body = decompress(
+    new Request(url, { redirect: 'follow' }),
+    new Response('hello world', {
+      status: 301,
+      headers: {
+        'content-encoding': 'gzip',
+        location: '/next-url'
+      }
+    })
+  )
+  assert.equal(await new Response(body).text(), 'hello world')
+})
+
+test('ignores HEAD requests', async () => {
+  const body = decompress(
+    new Request(url, { method: 'HEAD' }),
+    new Response('hello world', {
+      headers: {
+        'content-encoding': 'gzip'
+      }
+    })
+  )
+  assert.equal(await new Response(body).text(), 'hello world')
+})
+
+/**
+ * @note Undici does not support the "CONNECT" request method.
+ */
+test.skip('ignores CONNECT requests', async () => {
```

Comment:
We should, or what do you mean exactly?

Reply:
CONNECT is a special method which opens a tunnel.